README.md

```diff
@@ -1,4 +1,231 @@
+  ]
+});
+
+$config = [
+  'name' => 'group_1',
+  'title' => 'My Group',
+  'fields' => [
+    'MyProject/ACF/fields/Field_1'
+  ],
+  'location' => [
+    [
+      [
+      'param' => 'post_type',
+      'operator' => '==',
+      'value' => 'post',
+      ]
+    ]
+  ]
+];
+```
+
+The same can be done for the location.
+
+```php
+add_filter('MyProject/ACF/locations/PostTypePost', function ($location) {
+  return [
+    'param' => 'post_type',
+    'operator' => '==',
+    'value' => 'post',
+  ]
+});
+
+$config = [
+  'name' => 'group_1',
+  'title' => 'My Group',
+  'fields' => [
+    'MyProject/ACF/fields/Field_1'
+  ],
+  'location' => [
+    [
+      'MyProject/ACF/locations/PostTypePost'
+    ]
+  ]
+];
+```
+
+Combining the previous steps will yield the following result
+
+```php
+add_filter('MyProject/ACF/fields/Field_1', function ($field) {
+  return [
+    'label' => 'Sub Title',
+    'name' => 'sub_title',
+    'type' => 'text',
+  ]
+});
+
+add_filter('MyProject/ACF/locations/PostTypePost', function ($location) {
+  return [
+    'param' => 'post_type',
+    'operator' => '==',
+    'value' => 'post',
+  ]
+});
+
+$config = [
+  'name' => 'group_1',
+  'title' => 'My Group',
+  'fields' => [
+    'MyProject/ACF/fields/Field_1'
+  ],
+  'location' => [
+    [
+      'MyProject/ACF/locations/PostTypePost'
+    ]
+  ]
+];
+
+ACFComposer\ACFComposer::registerFieldGroup($config);
+```
+
+## API
+
+### ACFComposer\ACFComposer (class)
+
+#### registerFieldGroup (static)
+
+Main function of this package. Resolves a given field group config and registers an acf field group via `acf_add_local_field_group`.
+
+```php
+public static function registerFieldGroup(array $config)
+```
+
+### ACFComposer\ResolveConfig (class)
+
+#### forFieldGroup (static)
+
+Validate and generate a field group config from a given config array by adding keys and replacing filter strings with actual content.
+
+```php
+public static function forFieldGroup(array $config)
+```
+
+#### forField (static)
+
+Validate and generate a field config from a given config array by adding keys and replacing filter strings with actual content.
+
+```php
+public static function forField(array $config, array $parentKeys = [])
+```
+
+#### forLayout (static)
+
+Validate and generate a layout config from a given config array by adding keys and replacing filter strings with actual content.
+
+```php
+public static function forLayout(array $config, array $parentKeys = [])
+```
+
+#### forLocation (static)
+
+Validate a location config from a given config array.
+
+```php
+public static function forLocation(array $config)
+```
+
+## Maintainers
+
+This project is maintained by [bleech](https://github.com/bleech).
+
+Main people in charge of the repo are:
+
+- [Dominik Tränklein](https://github.com/domtra)
+- [Doğa Gürdal](https://github.com/Qakulukiam)
+
+## Contribute
+
+To contribute, please use github [issues](https://github.com/bleech/acf-field-group-composer/issues). PRs accepted.
```

Comment:
To contribute, please use github [issues](https://github.com/bleech/acf-field-group-composer/issues). **Pull requests are accepted.**